### DISCUSSION TITLE
Turn off character key short

### DISCUSSION THREAD

**Discussion by emalstm** [squidfunk/mkdocs-material#4273] [Q&A] [answered]
Hi,
Thank you so much for creating Material, I feel like my life has became a lot easier since we implemented it  a week ago ❤️ 

I see you have a lot of hidden a11y-features, which is great! Single character key shortcuts are very useful, but ideally the user should be able to turn them off. Is there a way to do that? I don't find it in the documentation.  

![image](https://user-images.githubusercontent.com/18416625/186127994-f16ad56a-e8fd-435c-b2cc-68e25288c7f3.png)
https://www.wuhcag.com/character-key-shortcuts/

**Comment by squidfunk** [comment, marked as answer]
Thanks for your kind words. There's currently no flag to turn it off, so the way to go would be to patch the theme and recompile it, removing the keys you don't want to bind. Note that keyboard navigation in search is something I definitely wouldn't recommend disabling, as users are used to how search engines like Google operate. For this reason, I'd consider it rather bad UX, which means that it's not a good fit for the project, as we try to provide the best possible standard UX.

**Reply by squidfunk** [reply]
If you want to [patch the theme](https://squidfunk.github.io/mkdocs-material/customization/#theme-development) – you just need to comment out or delete the following lines, and [build it](https://squidfunk.github.io/mkdocs-material/customization/#building-the-theme):

https://github.com/squidfunk/mkdocs-material/blob/acd154dfbd7672665bc07116ffb9e141a651d69b/src/assets/javascripts/bundle.ts#L139-L163

**Reply by humitos** [reply]
Hi! I found my question already answered here (basically, "there is no flag to turn this off"), however, since it's a 1 year old answer I wanted to know if this has changed. I would love to disable only the `global` hotkeys, I'm fine with `search` ones since they only work when there is focus on the search input. Thanks!

**Reply by squidfunk** [reply]
Nope, it did not change. We may evaluate in the future how and if we can make this configurable, but it is not a priority.

**Comment by Kiiv** [comment]
Hi,

interested in this too.
My documentation include some views from likec4 tool (https://playground.likec4.dev/w/tutorial/index/).
Those views contains a search box to find elements in diagrams but mkdocs does not detect that the focus is on a text area. 
For exemple, when I hit 'p', I'm redirected to the previous page in documentation.

**Reply by kamilkrzyskow** [reply]
Without a theme rebuild, this functionality can be cut out from the final `bundle.js` with a hook and proper replacement pattern similar to this hacky solution here:
- https://github.com/squidfunk/mkdocs-material/discussions/6844#discussioncomment-8670472

More recently, I've also used such hack on the [`mkdocs-nype`](https://nypesap.github.io/mkdocs-nype/) reference page with slight alterations:
- https://github.com/nypesap/mkdocs-nype/blob/main/docs_overrides/hooks/scope_currentcolor_css.py

**Comment by flefevre** [comment]
hello, 

did you introduce a way to deactivate shortcut in the configuration mkdocs.yml
i have noticed that for the theme mkdocs, there  is

```
theme:
  name: mkdocs
  shortcuts:
    help: 191    # ?
    next: 78     # n
    previous: 80 # p
    search: 83   # s
```

is it the case for materials?
and if we do not provide any item, does it deactivate it?

Thanks
